### PR TITLE
feat: create a span for traceability of tests

### PR DESCRIPTION
**Pull request by douglaz**
Now we now which test is generating which log, even if they run at the same time:
```
2023-09-14T21:17:28.790892Z  INFO on_chain_peg_in_and_peg_out_happy_case: fedimint_wallet_tests: Peg-in transaction claimed height=1415 tx=Transaction { version: 2, lock_time: PackedLockTime(1414), input: [TxIn { previous_output: OutPoint { txid: 37c7dbc03954476ac7f5e5990c550467f3b75ae72625e1df58d270bc2fa6235d, vout: 0 }, script_sig: Script(), sequence: Sequence(4294967293), witness: Witness { content: [64, 10, 230, 137, 46, 253, 53, 206, 47, 129, 55, 237, 34, 38, 34, 80, 127, 74, 175, 196, 42, 236, 87, 31, 210, 33, 170, 223, 182, 248, 54, 16, 150, 229, 253, 217, 201, 101, 110, 161, 136, 4, 168, 102, 36, 60, 242, 112, 217, 104, 30, 219, 58, 10, 6, 76, 29, 62, 68, 125, 183, 39, 156, 27, 11], witness_elements: 1, last: 0, second_to_last: 0 } }], output: [TxOut { value: 5000, script_pubkey: Script(OP_0 OP_PUSHBYTES_32 61443d44882b34a73e9e8661682f6f10c25f3b9a686f2eca78edf333119b5dc5) }, TxOut { value: 78062760, script_pubkey: Script(OP_PUSHNUM_1 OP_PUSHBYTES_32 6eeaf2ea46ee310f301975e9570d592711b17e9057cc8489021cddcb45ed2d89) }] }
2023-09-14T21:17:28.790965Z  INFO on_chain_peg_in_and_peg_out_happy_case: fedimint_wallet_tests: Peg-in finished for test on_chain_peg_in_and_peg_out_happy_case
2023-09-14T21:17:28.792862Z  INFO on_chain_peg_in_and_peg_out_happy_case: fedimint_wallet_server: Creating peg-out tx inputs=1 input_sats=5000 peg_out_sats=1000 fees_sats=730 fee_rate=1000 change_sats=3270
2023-09-14T21:17:28.792862Z  INFO on_chain_peg_in_and_peg_out_happy_case: fedimint_wallet_server: Creating peg-out tx inputs=1 input_sats=5000 peg_out_sats=1000 fees_sats=730 fee_rate=1000 change_sats=3270
2023-09-14T21:17:28.792910Z  INFO on_chain_peg_in_and_peg_out_happy_case: fedimint_wallet_server: Creating peg-out tx txid=2ae728b9c2c41279bea18f5fa5233b7ed5a296962765c1ba4a5f26cb92914e88
2023-09-14T21:17:28.792916Z  INFO on_chain_peg_in_and_peg_out_happy_case: fedimint_wallet_server: Creating peg-out tx txid=2ae728b9c2c41279bea18f5fa5233b7ed5a296962765c1ba4a5f26cb92914e88
2023-09-14T21:17:28.794953Z DEBUG on_chain_peg_in_and_peg_out_happy_case: timing: Operation timing name=autocmmit - commit_tx duration_ms=0
2023-09-14T21:17:29.029553Z DEBUG sanity_check_bitcoin_blocks: timing: Operation timing name=sign last epoch key duration_ms=3
2023-09-14T21:17:29.042543Z DEBUG sanity_check_bitcoin_blocks: timing: Operation timing name=sign last epoch key duration_ms=2
2023-09-14T21:17:29.045441Z DEBUG peg_outs_support_rbf: timing: Operation timing name=sign last epoch key duration_ms=2
2023-09-14T21:17:29.053717Z DEBUG peg_outs_support_rbf: timing: Operation timing name=sign last epoch key duration_ms=1
2023-09-14T21:17:29.242081Z  INFO sanity_check_bitcoin_blocks: consensus: 
```

Note it includes server module api calls.

### CODE REVIEW

File: modules/fedimint-ln-tests/tests/tests.rs [hunk 1]
```diff
@@ -124,7 +128,9 @@ async fn makes_internal_payments_within_federation() -> anyhow::Result<()> {
 #[tokio::test(flavor = "multi_thread")]
 async fn rejects_wrong_network_invoice() -> anyhow::Result<()> {
```
Comment (dpc) [inline modules/fedimint-ln-tests/tests/tests.rs:129]: Coincidentally I just started adding `#[instrument]` on every test and `[instrument(name = "<name>")]` on every binary `main` function.

We possibly could easily add a semgrep rule for that.

Comment (douglaz) [reply modules/fedimint-ln-tests/tests/tests.rs:129]: This is good, but for it to be really useful we need to propagate the `span` to the server side (i.e add .instrument() on some futures)

Comment (dpc) [reply modules/fedimint-ln-tests/tests/tests.rs:129]: I guess I don't understand spans and how to use them effectively.

I thought that I can slap `#[instrument]` in major places and job done. :D

Comment (douglaz) [reply modules/fedimint-ln-tests/tests/tests.rs:129]: In theory you should explicitly instrument every async function/code (https://docs.rs/tracing/latest/tracing/span/struct.Span.html#in-asynchronous-code).

But at least for most test purposes I found this is enough:
https://github.com/fedimint/fedimint/pull/3193/files#diff-476eca8e6f5bb7ac29c9bd2c6fbef5cff3c32b0f02ee2b7e24dffc7630eb7fb9R278

https://github.com/fedimint/fedimint/pull/3193/files#diff-6364ec89ecab2ba952666f453ab11412e5e0437503b88575358483403c20379bR120

File: modules/fedimint-wallet-tests/tests/tests.rs [hunk 2]
```diff
@@ -146,7 +148,9 @@ async fn sanity_check_bitcoin_blocks() -> anyhow::Result<()> {
 #[tokio::test(flavor = "multi_thread")]
 async fn on_chain_peg_in_and_peg_out_happy_case() -> anyhow::Result<()> {
     let fixtures = fixtures();
-    let fed = fixtures.new_fed().await;
+    let span = tracing::info_span!("on_chain_peg_in_and_peg_out_happy_case");
```
Comment (dpc) [inline modules/fedimint-wallet-tests/tests/tests.rs:151]: Why create a span, but then enter if after `new_fed`?

Comment (douglaz) [reply modules/fedimint-wallet-tests/tests/tests.rs:151]: :thinking:  You mean, why not enter it before? I guess I could do that. Perhaps add `#[instrument]` then create a span just for `new_fed`

**Comment by dpc** [comment]
OK. So running with `cargo nextest` is going to render this redundant, I think.

With `cargo nextest` each output of every test is captured separately, displayed only when needed, making figuring out which test is which simple. I'm going to convert everything to nextest soon, no matter the cost in property damages. Our CI insurance should cover it.

Also what I learned yesterday, wasting more time than I wanted playing with it is that:

https://github.com/tokio-rs/tracing/issues/2597

Also since spans don't propagete to spawned tasks, the're kind of clunky. And span name needs to be a constant making some dynamic stuff impossible.

I still don't understand log levels on spans, and I think chatgpt doesn't either.


Anyway - I think we should put it on hold for now, re-evalute after cargo nextest is fully rolled out.

**Comment by dpc** [comment]
@douglaz I think `cargo nextest` did indeed make identifying failing test not an issue, so closing.

**Pull request closed without merging**